ui.tests/test-module/specs/recaptcha/recaptcha.runtime.spec.js

```diff
@@ -112,35 +113,129 @@ describe("Form Runtime with Recaptcha Input", () => {
         })
     }
 
+    function updateRecaptchaSecretKey(secretKey) {
+        cy.openPage("/mnt/overlay/fd/af/cloudservices/recaptcha/properties.html?item=%2Fconf%2Fcore-components-it%2Fsamples%2Frecaptcha%2Fbasic%2Fsettings%2Fcloudconfigs%2Frecaptcha%2Fv2checkbox").then(x => {
+            cy.get('#recaptcha-cloudconfiguration-secret-key').clear().type(secretKey);
+            cy.get("#shell-propertiespage-doneactivator").click();
+        })
+    }
+
+
+    it("client side validation should fail if recaptcha is not filled", () => {
+        cy.previewForm(v2checkboxPagePath).then((p) => {
+            formContainer = p;
+        });
+        expect(formContainer, "formcontainer is initialized").to.not.be.null;
+        cy.get(`.cmp-adaptiveform-button__widget`).click().then(x => {
+            cy.get('.cmp-adaptiveform-recaptcha__errormessage').should('exist').contains('Please fill in this field.');
+        });
+    })
+
+    it("submission should pass for mandatory recaptcha v2", () => {
+        const secretKey="6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe";
```

Comment:
Why is secret key part of test case ? It should be in env variable

Reply:
Testing keys are provided by Google for recaptcha v2. So, there is no harm in adding them directly to the code base. 
https://developers.google.com/recaptcha/docs/faq#id-like-to-run-automated-tests-with-recaptcha.-what-should-i-do

For enterprise, we did not have any keys for testing provided by google, so we added it as a env variable.